packages/fxa-settings/src/components/DropDownAvatarMenu/index.tsx

```diff
@@ -0,0 +1,126 @@
+  useEscKeydownEffect(setRevealed);
+  const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
+  const dropDownId = 'drop-down-avatar-menu';
+
+  // TODO: FXA-2450
+  const [destroySession, { data, error }] = useMutation(
+    DESTROY_SESSION_MUTATION,
+    {
+      onError: (error) => {
+        revealAlertBar();
+        sentryMetrics.captureException(error);
+      },
+    }
+  );
+
+  const signOut = () => {
+    destroySession({
+      variables: { input: {} },
+    });
+  };
+
+  if (data) {
+    clearSignedInAccountUid();
+    window.location.assign(`${window.location.origin}/signin`);
+    return null;
+  }
+
+  return (
+    <>
+      {alertBarRevealed && error && (
+        <AlertBar onDismiss={hideAlertBar}>
+          <p data-testid="sign-out-error">Error text TBD. {error.message}</p>
+        </AlertBar>
+      )}
+      <div className="relative" ref={avatarMenuInsideRef}>
+        <button
+          onClick={toggleRevealed}
+          data-testid="drop-down-avatar-menu-toggle"
+          title="Firefox Account Menu"
+          aria-expanded={isRevealed}
+          aria-controls={dropDownId}
+          className="rounded-full border-2 border-transparent hover:border-purple-500 focus:border-purple-500 focus:outline-none active:border-purple-700 transition-standard"
+        >
+          <Avatar className="w-10 rounded-full" />
+        </button>
+        {isRevealed && (
+          <div
+            id={dropDownId}
+            data-testid={dropDownId}
+            className="drop-down-menu -left-52"
+          >
+            <div className="flex flex-wrap">
+              <div className="flex w-full p-4">
+                <div className="mr-3 flex flex-1">
+                  <Avatar className="w-10" />
+                </div>
+                <div className="flex flex-4 truncate">
+                  <p className="leading-5 max-w-full">
+                    <span className="text-grey-400 text-xs truncate">
+                      Signed in as
+                    </span>
+                    <span
+                      className="font-bold block max-w-full truncate"
+                      data-testid="drop-down-name-or-email"
+                    >
+                      {displayName || primaryEmail.email}
+                    </span>
+                  </p>
+                </div>
+              </div>
+              <div className="w-full">
+                <div className="bg-gradient-to-r from-blue-500 via-pink-700 to-yellow-500 h-px" />
+                <div className="p-6">
+                  <button
+                    className="pl-3 group"
+                    onClick={signOut}
+                    data-testid="sign-out"
+                  >
+                    <SignOut
+                      height="18"
+                      width="18"
+                      className="mr-3 inline-block stroke-current"
+                    />
+                    <span className="group-hover:underline">Sign out</span>
```

Comment:
Nitpick: this is ever so slightly improved with `vertical-align: middle;`

Reply:
_ever so slightly_ is like, less than 1px, this kind of cracked me up. But you're right, that less than 1px move does look slightly better 😂